### PR TITLE
Add internationalisation framework

### PR DESCRIPTION
**Pull request by TomAFrench**
There's been some interest in adding translations for languages other than english here: https://github.com/actualbudget/actual/discussions/199

I can't really work on this myself personally due time + language knowledge constraints but I'm leaving this PR open so that anyone can contribute translations to be merged into this PR and once we have more coverage then we can work on getting it merged into master (we'll need to add the ability to set the language in the UI, etc.).

For the time being you can choose which language to display by setting the `lng` flag in `src/locales/index.js`. We currently have very barebones i18n files which only handle a couple of strings on the bootstrapping page.

These files need to be filled out and  new ones created for any languages we want to support. To use these these translations, all the components in the app need to be updated in a similar fashion to the `Bootstrap` component (please use that component as a guide).

### CODE REVIEW

**Comment by TomAFrench** [comment]
@manuelcanepa There was a couple of issues which popped up which I've fixed:

1. There was a mismatch on the translation key used in the json file compared to in the React code.
2. Templated values need to be wrapped {{twice}} instead of {once}.
3. It seems like the `t()` function doesn't properly display `<strong>` tags so we need to use the <Trans> component in these cases.

If you run `yarn start` in the terminal then you should be able to check whether the translations appear correctly in the app.

**Comment by manuelcanepa** [comment]
Thanks for the corrections, im having troubles to start this project, so i make changes without running. Today i will try to start the environment to check every change that i makes.

**Comment by TomAFrench** [comment]
Ok, feel free to share any errors you're getting and we can see if we can work it out together.

**Comment by manuelcanepa** [comment]
@TomAFrench I dont have write access, thats why i can't contribute directly to this PR and open #219. BTW: I notice that `.editorconfig` says: 
```
[*]
indent_size = 2
```

And you indent json with 4 spaces, i propose you to add:
```
[*.json]
indent_size = 4
```

**Comment by manuelcanepa** [comment]
I think that this will be a headache to be translated: https://github.com/actualbudget/actual/blob/i18n/packages/loot-core/src/shared/schedules.js

**Comment by j-f1** [comment]
Working on that @manuelcanepa!

**Comment by manuelcanepa** [comment]
I see that the "es" language is fixed on config. @TomAFrench  What do you think about this: https://github.com/i18next/i18next-browser-languageDetector?

**Comment by TomAFrench** [comment]
I haven't put much thought into the language selection so far. We should definitely add a dropdown to the settings page but not sure if we need this package vs storing it in local storage ourselves.

File: packages/desktop-client/package.json [hunk 1]
```diff
@@ -42,6 +42,8 @@
     "fs-extra": "7.0.0",
     "glamor": "^2.20.40",
     "html-webpack-plugin": "4.0.0-alpha.2",
+    "i18next": "^21.9.1",
+    "i18next-parser": "^6.5.0",
```
Comment (manuelcanepa) [inline packages/desktop-client/package.json:46]: Why is on devDependencies?

**Comment by manuelcanepa** [comment]
What do we need to make this happend?

**Comment by MatissJanis** [comment]
Closing this one off as abandoned. But linking to this PR from an issue so if someone decides to pick this up again can refer back to this PR.

https://github.com/actualbudget/actual/issues/497

**Comment by vfaculteit** [comment]
I'd like to continue working on this, could you please reopen the i18n branch @MatissJanis? Also, for readability purposes we could directly add fallback translations, e.g., the string "Accounts" becomes t('general.accounts', "Accounts") in stead of t('general.accounts').

**Comment by manuelcanepa** [comment]
> I'd like to continue working on this, could you please reopen the i18n branch @MatissJanis? Also, for readability purposes we could directly add fallback translations, e.g., the string "Accounts" becomes t('general.accounts', "Accounts") in stead of t('general.accounts').

I like it, how this works with [interpolation](https://www.i18next.com/translation-function/interpolation)?

**Comment by vfaculteit** [comment]
> > I'd like to continue working on this, could you please reopen the i18n branch @MatissJanis? Also, for readability purposes we could directly add fallback translations, e.g., the string "Accounts" becomes t('general.accounts', "Accounts") in stead of t('general.accounts').
> 
> I like it, how this works with [interpolation](https://www.i18next.com/translation-function/interpolation)?

This should work: e.g., `\`Account: ${accountName}\`` becomes `t('general.accountName', {name: accountName, defaultValue: 'Account: {{name}}'}`

**Comment by fabienheureux** [comment]
@vfaculteit did you start working on this ? 
I might be interested to help as well, French speaker here, with experience in translating react projects. 
I don't want to duplicate the effort, let me know if there are areas where you think you might need some help.

**Comment by JeroenRobben** [comment]
@fabienheureux Yes I did - I'm currently working in this [repo](https://github.com/vfaculteit/actual/tree/i18n) (branch 'i18n'). I took the approach to just use the translated word or sentence as a key, i.e., for the example above, "Accounts" becomes t('Accounts') in stead of t('general.accounts') or t('general.accounts', "Accounts"), except for some complex cases, e.g., when interpolation is used. I can give you access to this repo, although it would be nicer to be able to just continue here (@MatissJanis  would it be possible to reopen the i18n branch?)

**Comment by fabienheureux** [comment]
Great, glad to see the work is in progress ! 
No rush in getting access to the repo, I can follow from there and check where I can be of any help, and base some work on this branch. 

Are there any blockers I could help with ?

**Comment by rafacla** [comment]
Hello guys, brazilian here, can help translating to pt-BR

**Comment by lelemm** [comment]
I can help too

> Hello guys, brazilian here, can help translating to pt-BR

**Pull request closed without merging**